hail/src/main/scala/is/hail/expr/types/physical/PType.scala

```diff
@@ -143,8 +144,6 @@ object PType {
       case PVoid => PVoid
     }
   }
-
-  def canonical(t: EncodedType): PType = canonical(t.virtualType)
```

Comment:
I think this is a good thing to have, no?

Reply:
e.g. requiredness will be in both ETypes and PTypes, but not Types. So if you construct a canonical physical type from a virtual type, it must be all-optional.